src/main/java/Deadline.java

```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by + ")";
+    }
```

Comment:
Would it be better if this variable was named something more descriptive, like deadlineDate or completeBy?